content/articles/rust-for-web-services.md

```diff
@@ -0,0 +1,134 @@
+Title: Trying Rust for web services
+Date: 2015-09-30
+Short_summary: I made a very basic Rust backend, here's what I think about it
+Category: Dev
+Authors: Vincent
+
+*Reading time: ~15 minutes.*
+
+
+Developing web apps in dynamic languages is a breeze when using frameworks like [Django](https//www.djangoproject.com/) for Python.
+The downsides are that software written in dynamic languages is harder, at least in my opinion, to maintain, to refactor and you also need to write tests to cover potential errors that would simply not be possible with a compiler.
+<!-- PELICAN_END_SUMMARY -->
+
+
+While I would use Flask or Django for a small project, I would definitely prefer having help from a compiler for a long-lived product. The fact that compiled languages are generally faster than dynamic ones is a nice bonus too.
+
+The two choices in my mind right now for a compiled language are Go and Rust. Some (ie Tom) would say Haskell, others would say Scala but in the end it's down to preferences.
+
+I previously used Go for a few projects and I quite liked it despite some annoying quirks such as package management (might be solved by [gb](https://getgb.io/) now) and seeing `interface {}` in some libraries to get around the weak type system. On the other hand, I have been following [Rust](https://www.rust-lang.org/) developement for quite a while but didn't play with it more than a couple of toy programs. I was keen to see if it could work as the backend for a web app, which in short means a HTTP server receiving and sending JSON while talking to a Postgres database. There are obviously an awful lot of other things but let's keep it simple for now.
+
+
+## The demo
+You can see the project at [https://github.com/Keats/webrust](https://github.com/Keats/webrust).
```

Comment:
Opinion follows :

The code looks reasonable, rust is quite clear.
On the minus side the code looks a bit low level (there are many lines to set-up the database router, etc.) for a web application.
I might be biased with my lisp and C experience but macros can easily be abused, let's hope the rust culture doesn't fall in that trap too much!
I'm not sure about the `#![feature(custom_derive, plugin)]` types of directives.